File: Bridge/Pasternak/Shape/Circle.java
package Inheritance.Shape;

import Inheritance.Color.Color;

public class Circle extends Shape {
    double radius;

    public Circle(double radius, Color color) {
        super(color);
        this.radius = radius;
    }

    public double getPerimeter() {
        return 2.0 * this.radius * Math.PI;
    }

    public double getArea() {
        return Math.pow(this.radius, 2.0) * Math.PI;
    }
    @Override
    public String toString() {
        return "Сircle{radius=" + radius + ", perimeter=" + getPerimeter() + ", area=" + getArea() + ", color=" + this.color + "}";
    }
}

File: Bridge/Trushchak-03/Bridge/Bridge/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bridge
{
    class Circle : Shape
    {
        private double radius;

        public Circle(double radius, IColor color) : base(color)
        {
            this.radius = radius;
        }

        public override double GetArea()
        {
            try
            {
                if (radius <= 0)
                    throw new ArgumentException("Radius of a circle must be positive.");

                return Math.PI * radius * radius;
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine($"Error calculating area: {ex.Message}");
                return 0;
            }
        }

        public override double GetPerimeter()
        {
            try
            {
                if (radius <= 0)
                    throw new ArgumentException("Radius of a circle must be positive.");

                return 2 * Math.PI * radius;
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine($"Error calculating perimeter: {ex.Message}");
                return 0;
            }
        }

        public override void Draw()
        {
            Console.WriteLine($"Drawing a {color.GetColor()} circle with radius {radius}");
        }
    }


}
